src/allotropy/allotrope/schema_parser/generate_schemas.py

```diff
@@ -13,30 +16,29 @@
 
 from allotropy.allotrope.schema_parser.backup_manager import (
     backup,
+    backup_paths,
     is_backup_file,
     is_file_changed,
-    restore_backup,
 )
 from allotropy.allotrope.schema_parser.model_class_editor import modify_file
 from allotropy.allotrope.schema_parser.path_util import (
-    CUSTOM_MODELS_PATH,
     GENERATED_SHARED_PATHS,
-    get_model_file_from_schema_path,
+    get_model_path_from_schema_path,
     get_rel_schema_path,
     MODEL_DIR_PATH,
     SCHEMA_DIR_PATH,
-    UNITS_MODELS_PATH,
 )
 from allotropy.allotrope.schema_parser.schema_cleaner import SchemaCleaner
 from allotropy.allotrope.schema_parser.update_units import update_unit_files
+from allotropy.allotrope.schemas import get_schema
 
 
 def lint_file(model_path: Path) -> None:
     # The first run of ruff changes typing annotations and causes unused imports. We catch failure
     # due to unused imports.
     try:
         subprocess.check_call(
-            f"ruff {model_path} --fix",
+            f"ruff --ignore N999 {model_path} --fix",
```

Comment:
why do we need this?

Reply:
Added a comment. N999 checks that the filename is valid for a python module, which fails here because we're now calling with the backup file (which has .uuid.bak in the path)